radis/io/exomol.py

```diff
@@ -719,191 +701,57 @@ def __init__(
-                        trans = exomolapi.read_trans(trans_file, engine="csv")
-                        trans.to_feather(trans_feather_path)
-                        #!!TODO:restrict NOW the trans size to avoid useless overload of memory and CPU
-                        # trans = trans[(trans['nu'] > self.nurange[0] - self.margin) & (trans['nu'] < self.nurange[1] + self.margin)]
-                    ndtrans = trans.to_numpy()
-                    # mask needs to be applied   (in feather mode we don't sleect wavneumbers)
-                    mask_needed = True
-                elif engine == "vaex":
-                    if trans_file.with_suffix(".hdf5").exists():
-                        trans = vaex.open(trans_file.with_suffix(".hdf5"))
-                        cdt = 1
-                        if not np.isneginf(self.nurange[0]):
-                            cdt *= trans.nu_lines > self.nurange[0] - self.margin
-                        if not np.isinf(self.nurange[1]):
-                            cdt *= trans.nu_lines < self.nurange[1] - self.margin
-                        if not np.isneginf(self.crit):
-                            cdt *= trans.Sij0 > self.crit
-                        if cdt != 1:
-                            trans = trans[cdt]
-
-                        # mask has been already applied
-                        mask_needed = False
-                    else:
-                        print(
-                            "Note: Caching line transition data to the HDF5 format with vaex. After the second time, it will become much faster."
-                        )
-                        trans = exomolapi.read_trans(trans_file, engine="vaex")
-
-                        # Check validity
-                        b = trans.nu_lines.isnan()  # TODO: check if can be made faster?
-                        if b.sum() > 0:
-                            raise NotImplementedError(
-                                "transition wavenumber not given. Has to be computed from upper/lower level energy look up"
-                            )
-
-                        # TODO Load extra columns (precomputed Sij0, etc.)
-                        # trans_file_extra = self.trans_file.with_name(
-                        #    self.trans_file.stem + "_extra" + self.trans_file.suffix
-                        # )
-                        # if trans_file_extra.exists():
-                        #    trans_extra = vaex.open(trans_file_extra)
-                        #    trans = trans.join(
-                        #        trans_extra
-                        #    )  # merge without column name will merge on rows basis
-
-                        # mask needs to be applied
-                        mask_needed = True
-
-                    ndtrans = vaex.array_types.to_numpy(trans)
                 self.trans_file.append(trans_file)
+                self.num_tag.append(dic_def["numtag"][i])
+
+        # Look-up missing parameters and write file
```

Comment:
Separate Find Filenames (1st) and Download & write them (after) --> helps remove copies of code, more readable